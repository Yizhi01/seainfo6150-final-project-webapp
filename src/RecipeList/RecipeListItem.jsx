import React from "react";
import { Link } from "react-router-dom";
import styles from "./RecipeListItem.module.css";
import PropTypes from "prop-types";

const RecipeListItem = (props) => {
  return (
    <div>
      <Link to={"/recipe/" + props.id} className={styles.recipeListItem} >
      <img
          className={styles.recipeImage}
          src={props.url}
          alt={props.name}
        />
        <p className={styles.recipeName}>{props.name}</p>
      </Link>
    </div>
  )
}

RecipeListItem.propTypes = {
  id: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
};

export default RecipeListItem;