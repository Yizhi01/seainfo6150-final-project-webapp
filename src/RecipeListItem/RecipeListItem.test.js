import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import RecipeListItem from "./RecipeListItem.jsx";

it("renders the RecipeListItem component correctly", () => {
  const { container } = render(
    <Router>
      <RecipeListItem id="1" name="Fruit Smoothie" url="https://source.unsplash.com/CpBBsda2eRI/750x1000" />
    </Router>
   
  );
  expect(container).toMatchSnapshot();
});