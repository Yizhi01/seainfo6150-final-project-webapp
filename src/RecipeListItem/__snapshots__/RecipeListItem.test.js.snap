// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the RecipeListItem component correctly 1`] = `
<div>
  <div>
    <a
      class="recipeListItem"
      href="/recipe/1"
    >
      <img
        alt="Fruit Smoothie"
        class="recipeImage"
        src="https://source.unsplash.com/CpBBsda2eRI/750x1000"
      />
      <p
        class="recipeName"
      >
        Fruit Smoothie
      </p>
    </a>
  </div>
</div>
`;
